Samples/Docker-Simple/Silo/Program.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.Extensions.Logging;
+using Orleans.Hosting;
+using Orleans.Runtime.Configuration;
+using System;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Silo
+{
+    class Program
+    {
+        private static ISiloHost silo;
+        private static readonly AutoResetEvent closing = new AutoResetEvent(false);
+
+        static void Main(string[] args)
+        {
+            var connectionString = File.ReadAllText("connection-string.txt");
+
+            var config = new ClusterConfiguration();
+            config.Globals.DataConnectionString = connectionString;
+            config.Globals.LivenessType = GlobalConfiguration.LivenessProviderType.AzureTable;
+            config.Globals.ReminderServiceType = GlobalConfiguration.ReminderServiceProviderType.AzureTable;
+            config.Globals.FastKillOnCancelKeyPress = true;
+            config.Defaults.Port = 11111;
+            config.Defaults.ProxyGatewayEndpoint = new IPEndPoint(IPAddress.Any, 30000);
+
+            silo = new SiloHostBuilder()
+                .AddApplicationPartsFromAppDomain()
+                .AddApplicationPartsFromBasePath()
+                .UseConfiguration(config)
+                .ConfigureLogging(builder => builder.SetMinimumLevel(LogLevel.Warning).AddConsole())
+                .Build();
+
+            Task.Run(StartSilo);
+
+            AppDomain.CurrentDomain.ProcessExit += (object sender, EventArgs e) =>
+            {
+                Console.WriteLine("ProcessExit fired");
+                Task.Run(StopSilo);
```

Comment:
Is this correct? If you don't block ProcessExit, I believe it will close immediately, right?

Reply:
You may be right. I will just use a sync call here. 

And use `AssemblyLoadContext.Default.Unloading`, I put previsouly ProcessExit earlier for test purpose